python/lib/core/dmod/test/test_context.py

```diff
@@ -0,0 +1,1599 @@
+"""
+Unit tests used to ensure that dmod.core.context operations behave as intended
+"""
+from __future__ import annotations
+
+import abc
+import dataclasses
+import inspect
+import os
+import sys
+import typing
+import unittest
+import multiprocessing
+import string
+import random
+import re
+
+from collections import namedtuple
+from itertools import permutations
+from multiprocessing import managers
+from multiprocessing import pool
+
+from typing_extensions import ParamSpec
+from typing_extensions import Self
+
+from ..core import context
+
+VARIABLE_ARGUMENTS = ParamSpec("VARIABLE_ARGUMENTS")
+
+T = typing.TypeVar("T")
+R = typing.TypeVar("R")
+
+STATIC_METHOD_ONE_VALUE = 95
+STATIC_METHOD_TWO_VALUE = "blue"
+
+CLASS_METHOD_ONE_VALUE = False
+CLASS_METHOD_TWO_VALUE = 3+3j
+
+STRING_INTEGER_PATTERN = re.compile(r"-?\d+")
+MEMBER_SPLIT_PATTERN = re.compile(r"[./]")
+
+MutationTuple = namedtuple("MutationTuple", ["field", "value", "should_be_equal"])
+
+
+def shared_class_two_instance_method_formula(*args) -> int:
+    """
+    The formula for the instance method for SharedClassTwo
+
+    Args:
+        *args: The values to include in the formula
+
+    Returns:
+        An integer representing the size of passed in values
+    """
+    total: int = 0
+
+    for item in args:
+        if isinstance(item, typing.Sized):
+            total += len(item)
+        else:
+            total += item
+
+    return total
+
+
+def make_word(min_length: int = None, max_length: int = None, character_set: str = None, avoid: str = None) -> str:
```

Comment:
Unless we are doing fuzz testing, its generally not the best idea to test using random data. Perhaps we use something like this to automate the process of creating random data that we use to drive a table driven test, but this just feels like the beginning of a flakey test waiting to happen.

Reply:
This is used in a way that's almost like a table driven test. This is used to generate values that make up 10s of identity mutation tests. The value does not generally as much as "was this the same before and after mutation?"

Random words and random numbers is the easiest way to generate that amount of data.

Its usage is involved in mass parameter generation for individual test generation. These values flow into others and when assertions are made, they result in messages like: "#{step_number})'{step_result.step_name}' failed - expected '{step_result.expected_result}' but resulted in '{step_result.value}'", or:

```
[Test SharedClassTwo] Test Equality

#3) 'Use 'set_c' to set 'c' to 's79SD8' on Partially Mixed Class Two' - expected 's79SD8' but resulted in 'fUsd8Sl8'
```